app/models/question.rb

```diff
@@ -0,0 +1,50 @@
+class Question < ApplicationRecord
+  include ::TokenHandler
+
+  belongs_to :user
+
+  before_create -> { generate_token(:published_token) }
+
+  before_save :ensure_published_question_cannot_be_drafted
+
+  with_options if: :published? do
+    validates :title, presence: true, uniqueness: true
+    validates :content, presence: true, length: { minimum: 15 }
+    validates :topic_list, presence: true
+  end
+
+  has_one_attached :pdf_attachment
```

Comment:
it should not be pdf_attachment, as filr can be of any type

Reply:
it was written that file can be pdf only